### PR TITLE
CASMPET-5081 Add spire upgrade script

### PR DESCRIPTION
**Pull request by kburns-hpe**
# Description

This adds the upgrade script for spire. This will require the spire-agent RPMs be uploaded to a csm rpm repo that's setup in zypper prior to being run.

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [X] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by leliasen-hpe — APPROVED** [review]

**Comment by kburns-hpe** [comment]
The jenkins pr-merge job is failing due to "No space left on device"

File: upgrade/scripts/upgrade/util/upgrade-spire.sh [hunk 1]
```diff
@@ -22,25 +22,82 @@
...
+  sshnh "$node" zypper install -y spire-agent
+
+  echo "$node: Starting spire-agent"
+  sshnh "$node" systemctl start spire-agent
+done
+
+echo "Uninstalling spire helm chart"
+helm uninstall -n spire spire
+echo "Uninstalling spire PVCs"
+kubectl delete -n spire pvc spire-data-spire-server-0 spire-data-spire-server-1 spire-data-spire-server-2
```
Comment (rustydb) [inline upgrade/scripts/upgrade/util/upgrade-spire.sh:53]: Does this need to fetch every spire-data server or will there always ever be 3?

File: upgrade/scripts/upgrade/util/upgrade-spire.sh [hunk 2]
```diff
@@ -22,25 +22,82 @@
...
+for chart in $charts; do
+  if [[ $chart != "spire" ]]; then
+    yq d -i $BUILDDIR/spireupgrade.yaml "spec.charts.(name==$chart)"
+  fi
+done
+
+yq w -i $BUILDDIR/spireupgrade.yaml "metadata.name" "spireupgrade"
+yq d -i $BUILDDIR/spireupgrade.yaml "spec.sources"
+
+loftsman ship --charts-path "${CSM_ARTI_DIR}/helm/" --manifest-path $BUILDDIR/spireupgrade.yaml
```
Comment (mharding-hpe) [inline upgrade/scripts/upgrade/util/upgrade-spire.sh:85]:
```suggestion
charts="$(yq r "$BUILDDIR/spireupgrade.yaml" 'spec.charts[*].name')"
for chart in $charts; do
  if [[ $chart != "spire" ]]; then
    yq d -i "$BUILDDIR/spireupgrade.yaml" "spec.charts.(name==$chart)"
  fi
done

yq w -i "$BUILDDIR/spireupgrade.yaml" "metadata.name" "spireupgrade"
yq d -i "$BUILDDIR/spireupgrade.yaml" "spec.sources"

loftsman ship --charts-path "${CSM_ARTI_DIR}/helm/" --manifest-path "$BUILDDIR/spireupgrade.yaml"
```

File: upgrade/scripts/upgrade/util/upgrade-spire.sh [hunk 3]
```diff
@@ -22,25 +22,82 @@
...
+  else
+    echo "spire-server did not start after $(echo "$RETRY_SECONDS" \* "$MAX_RETRIES" | bc) seconds."
+    exit 1
+  fi
+  sleep "$RETRY_SECONDS"
+done
 
-deploySpire
+echo "Joining storage nodes to spire"
+/opt/cray/platform-utils/spire/fix-spire-on-storage.sh
```
Comment (mharding-hpe) [inline upgrade/scripts/upgrade/util/upgrade-spire.sh:103]: Should the exit code of this be checked? I know that if it exits non-0, this overall script will exit non-0, but perhaps we'd also want to print an error message explaining what failed?

**Comment by mharding-hpe** [comment]
I assume this will also need a main backport once it's ready -- is that right?

**Comment by kburns-hpe** [comment]
This is being removed as the spire upgrade is going in a different direction.

**Pull request closed without merging**